### DISCUSSION TITLE
Is it efficient to use tarred data with bucketing for single machine multi-gpu ASR training?

### DISCUSSION THREAD

**Discussion by mehadi92** [NVIDIA-NeMo/NeMo#6750] [Q&A] [answered]
For ASR training we can not use only data bucketing. We must convert it to tarred data. My question is will tarred data + data bucketing make training faster for single machine + multi GPU? 

Or tarred data + data bucketing only make training faster for multi-node training only?

**Comment by titu1994** [comment, marked as answer]
Tarred dataset can be used even for single machines but it won't give significant benefit there. Bucketing will make it faster to train even on single node cause it samples files with specific durations in each batch but it's main speedup is on using multi node.
